### PR TITLE
split data without making a copy.

### PR DESCRIPTION
**Pull request by hguan6**
The Sklearn train_test_split makes a copy of the data, which requires a very big memory. The change in this commit is to use a reference instead of a copy.

### CODE REVIEW

File: model-inference/decisionTree/experiments/data_processing.py [hunk 1]
```diff
@@ -239,10 +241,10 @@ def create_tables(
         raise ValueError(f"{dataset} not supported")
 
     # Split dataset
-    train, test = train_test_split(df, random_state=77, test_size=datasetconfig["test"])
-    del df
-    gc.collect()
-
+    # df = shuffle(df, random_state=77)
```
Comment (venkate5hgunda) [inline model-inference/decisionTree/experiments/data_processing.py:244]: Hey Hong, do you mean to uncomment this? We would like the split to be random, right?

**Review by venkate5hgunda — COMMENTED** [review]
Please check if the data needs to be shuffled. Otherwise, LGTM.

File: model-inference/decisionTree/experiments/data_processing.py (hunk 1 above)
Comment (hguan6) [reply model-inference/decisionTree/experiments/data_processing.py:244]: Yes. We can discuss this point.  I think it is okay not to shuffle the data. The shuffling operation uses more memory, which may cause memory issue for large datasets.

Comment (jiazou-bigdata) [reply model-inference/decisionTree/experiments/data_processing.py:244]: Will shuffling affect accuracy? We want to make the benchmark realistic by working with good quality models.
If it won't affect accuracy, e.g., if the training process will also shuffle the data, we can definitely remove the operator to improve the performance of our benchmark pipeline.
